docs/upgrades.md

```diff
@@ -0,0 +1,318 @@
+Upgrades and Installing Particular Versions
+===========================================
+
+When installing Pulp, there are 3 software components whose versions you should be mindful of:
+1. pulp_installer
+2. pulpcore
+3. 1 or more plugins
+
+If all you want to do is install Pulp and upgrade it to the very latest version in the future,
+you may not need to specify any versions. However, it is recommended to specify versions at the time
+of installation for 2 reasons
+* Handling a situation where not all your plugins compatible with the latest pulpcore version.
```

Comment:
do we have a document with a compatibility matrix so that users can identify which versions they can provide/configure to install/upgrade?
for example, what if a user wants to install pulp_pluginA version A.B.C and pulpcore D.E.F but pulpcore:D.E.F is not compatible with pulp_pluginA:A.B.C? I understand that the installer will not proceed with the installation, but how or where does someone can find which version can be provided to the installer?

Reply:
They have to check the READMEs, the release announcements, etc. As mentioned in the docs. For now (see comment about PDC.)